judge/views/api/api_v2.py

```diff
@@ -257,6 +257,8 @@ def get_object_data(self, contest):
             'rankings': [
                 {
                     'user': participation.username,
+                    'start_time': participation.start.isoformat(),
```

Comment:
I'm not sure how Django handles fetching related objects through an object, so we need to make sure that we're not fetching `contest` for each participation (there could be > 200 participations in a single contest resulting with > 200 queries).

Also, we might not even need to `select_related`, since we can just overwrite the `contest` attribute with the `contest` we already have (since the contest doesn't change for any of these participations).

Reply:
Based on some testing, seems like the way django handles many-to-one relationships is by loading their dependency from the db. Meaning it would result in > 200 queries if there was > 200 participations in a single contest.

I think `select_related` would be the correct option as modifying the `contest` attribute seems a little sketch.

Reply:
Just set `.contest` and be done with it.